common/workunit/workunit.cpp

```diff
@@ -2487,8 +2487,73 @@ class CompoundAggregator : implements StatisticAggregator
     IArrayOf<StatisticAggregator> aggregators;
 };
 
+class CostAggregator : public StatisticAggregator
+{
+public:
+    CostAggregator(StatsAggregation & _summary, bool _excludehThor) : StatisticAggregator(StCostExecute), summary(_summary), excludehThor(_excludehThor) {}
+
+    virtual void noteStatistic(StatisticKind kind, unsigned __int64 value, IConstWUStatistic & extra) override
```

Comment:
This code is confusing for several reasons:
- haveGraphCost is calculated, but never used.
- It will not work for roxie (when roxie is run in per-workunit mode)
- Efficiency probably isn't a big issue, but in other situations using nextSibling() would be much more efficient.
- excludehThor looks like it has been misspelled.

Reply:
With the suggested change to workflow item costs, this logic should be removed.  There isn't a need for this class.  Something like the following instead:

```
  Owned<IConstWUScopeIterator> it = &wu->getScopeIterator(filter);
  for (it->first(); it->isValid(); )
    {
        if (it->getStat(CostExecute, value)) // I think this should always be true
       {  sum+= value; it->nextSibling(); }
        else
            it->next();
    }
    return sum;

```

Reply:
Sometimes the children of a scope is included/excluded based on the creator (hThor or thor) so we need to be able to distinguish the creator (to determine if it's an hthor stat or thor stat).  

The only way to do that is at present within the visitor class.